### ISSUE TITLE
Unpublished versions can be selected on the publication page

### ISSUE THREAD

**Issue by cllde8** [jochenklar/chemotion_REPO#7]
Unpublished versions can be selected on the publication page.

Please follow these steps:

Create a new version for a sample from the "Data Publications" page. You will see a tooltip stating, "A new version of this sample has already been created."

[Sample](https://chemotion.jochenklar.dev/home/publications/molecules/847)

![screenshot-chemotion_jochenklar_dev-2024_07_22-14_54_45](https://github.com/user-attachments/assets/aab6d3ea-1243-417d-a420-cd15eadb58a8)

Go to the "Review" page, where the status of this sample is "pending," indicating that it is not published yet.

Then, go to the "Data Publications" page, click on the reaction, and scroll down to the product section. Here, the user can select the unpublished version.

[Reaction](https://chemotion.jochenklar.dev/home/publications/reactions/69)

![screenshot-chemotion_jochenklar_dev-2024_07_22-14_56_47](https://github.com/user-attachments/assets/8aa70ed6-f5c4-4b50-abac-4f843c9f3df5)

Could you please take a look?

Thank you.

**Comment by jochenklar** [comment]
Fixed in 15e359d2b8e4654c4d8489f589c3ca21b8a7908a. Please check the code, this might be done more efficient/elegant:

```ruby
        p[:versions] = pub_product_tag['versions'].reduce([]) do |versions, version_id|
          sample = Sample.find(version_id)
          if sample.publication.state == 'completed'
            versions.append Entities::SampleEntity.represent(sample, serializable: true)
          end
          versions
        end
```

**Comment by cllde8** [comment]
Thanks a lot, it works.
